__tests__/app.test.js

```diff
@@ -300,3 +301,34 @@ describe('PATCH /api/articles/:article_id', () => {
         })
     });
 });
+
+describe('GET api/users', () => {
+    test('resolves with status 200 and list of all users with correct key value pairs', () => {
+        return request(app).get('/api/users').expect(200).then((res) => {
+            const users = res.body.users;
+            expect(users).toHaveLength(4);
+            expect(users).toEqual(expect.arrayContaining([expect.objectContaining({
+                username: 'butter_bridge',
+                name: 'jonny',
+                avatar_url:
+                'https://www.healthytherapies.com/wp-content/uploads/2016/06/Lime3.jpg'
+            })]))
+            expect(users).toEqual(expect.arrayContaining([expect.objectContaining({
+                username: 'icellusedkars',
+                name: 'sam',
+                avatar_url: 'https://avatars2.githubusercontent.com/u/24604688?s=460&v=4'
+            })]))
+            expect(users).toEqual(expect.arrayContaining([expect.objectContaining({
+                username: 'rogersop',
+                name: 'paul',
+                avatar_url: 'https://avatars2.githubusercontent.com/u/24394918?s=400&v=4'
+            })]))
+            expect(users).toEqual(expect.arrayContaining([expect.objectContaining({
+                username: 'lurker',
+                name: 'do_nothing',
+                avatar_url:
+                'https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png'
+            })]))
```

Comment:
This is extremely thorough, I wonder though... if you're going to test for each object of the array could you not make one assertion of `expect(users).toEqual([... <the whole array> ...])` ?
On the other hand what you have here is open to expansion should a new user be added to the test data, the test won't break. We wouldn't usually expect the test data to change but it's interesting to weigh the two approaches up